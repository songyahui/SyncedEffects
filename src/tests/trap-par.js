"use hopscript"

var hh = require("hiphop");

var prg = <hh.module A B C>
  <hh.trap T>
    <hh.parallel>
      <hh.sequence>
	<hh.emit A/>
	<hh.exit T/>
      </hh.sequence>
      <hh.sequence>
	<hh.emit B/>
	<hh.pause/>
	<hh.emit C/>
      </hh.sequence>
    </hh.parallel>
  </hh.trap>
</hh.module>;

exports.prg = new hh.ReactiveMachine(prg, "trappar");
