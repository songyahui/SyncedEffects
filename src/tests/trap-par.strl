module trappar:

output A, B, C;

trap T in
        [ emit A;
          exit T ]
    ||
        [ emit B;
          pause;
          emit C ]
end trap

end module