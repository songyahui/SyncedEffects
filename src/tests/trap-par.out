trappar> ;
--- Output: A B
trappar> ;
--- Output:
trappar> ;
--- Output:
trappar> ;
--- Output:
